### PR TITLE
HHH-9933 Uninitialized PersistentSet with lazy="extra" doesn't respect added elements

### PR DESCRIPTION
**Pull request by racon-am**
HHH-9933 Pull request for https://hibernate.atlassian.net/browse/HHH-9933

### CODE REVIEW

**Comment by brmeyer** [comment]
This really needs cleaned up before we could consider it.

1.) Not sure what JRAPSOA-437 references -- should be HHH-9933
2.) This all needs squahsed down into a single commit.
3.) Your `repositories` changes in *.gradle need reverted.

File: hibernate-core/src/test/java/org/hibernate/test/extralazy/ExtraLazyTest.java [hunk 1]
```diff
@@ -261,5 +261,169 @@ public void testMap() {
 		assertNotNull(child2);
 		session2.close();
 	}
+
```
Comment (brmeyer) [inline hibernate-core/src/test/java/org/hibernate/test/extralazy/ExtraLazyTest.java]: Everyone one of these tests has an identical preparation step.  Pull that out into a common method?

**Comment by richardfontana** [comment]
@vladmihalcea this could be integrated without requiring the CLA.

**Comment by vladmihalcea** [comment]
Superceded by this [PR](https://github.com/hibernate/hibernate-orm/pull/2056).

**Pull request closed without merging**